superset/assets/javascripts/explorev2/components/ExploreViewContainer.jsx

```diff
@@ -33,12 +33,11 @@ class ExploreViewContainer extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    let refreshChart = false;
-    autoQueryFields.forEach((field) => {
-      if (nextProps.form_data[field] !== this.props.form_data[field]) {
-        refreshChart = true;
-      }
-    });
+    const refreshChart = Object.keys(nextProps.form_data).some((field) => (
+      nextProps.form_data[field] !== this.props.form_data[field]
+      && (typeof nextProps.form_data[field] === 'boolean'
```

Comment:
Does this mean you assume that all boolean fields should act like an `autoQueryFields`? It may not be the case. I think you probably want to add these fields in the `autoQueryFields` array instead.